sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/CSVOptions.scala

```diff
@@ -253,16 +253,18 @@ class CSVOptions(
   /**
    * A string between two consecutive JSON records.
    */
-  val lineSeparator: Option[String] = parameters.get(LINE_SEP).map { sep =>
-    require(sep.nonEmpty, "'lineSep' cannot be an empty string.")
-    // Intentionally allow it up to 2 for Window's CRLF although multiple
-    // characters have an issue with quotes. This is intentionally undocumented.
-    require(sep.length <= 2, "'lineSep' can contain only 1 character.")
-    if (sep.length == 2) logWarning("It is not recommended to set 'lineSep' " +
-      "with 2 characters due to the limitation of supporting multi-char 'lineSep' within quotes.")
-    sep
+  val lineSeparator: Option[String] = parameters.get(LINE_SEP) match {
+    case Some(sep) if sep != null =>
```

Comment:
Is a null line separator even valid? I'd imagine that should be an error, if an empty one is

Reply:
As far as my understanding, `None` when passed through python (`lineSep: None)` ends up being `null` in this function. Hence, I have a check for null.

Reply:
None isn't the default, right? if a user passes None, it feels like that should be an error

Reply:
@srowen I can see that it should be an error just like an empty value `''`. 
However, one of the use cases where this might be helpful is where someone might need to dynamically switch the value for `lineSep` from `None` to something else. For example, code generation is one example that comes to mind, where a base template with `lineSep: None` is used that can be replaced with appropriate values as desired.

Reply:
Hm, but generated code would not need to set 'None' right? that isn't meaningful at the point you actually use Spark

Reply:
I see your point. I guess we could also use another placeholder value instead of `None` in that scenario. 
I can revert the changes and add a `require` clause for null check instead of letting it through. Or would you rather have it throw an `NullPointerException`?

Reply:
I'd just require() it for consistency, unless there's an argument for other handling. (If there is, it'd probably apply to the "" case too)

Reply:
Done.